### PR TITLE
[blog] Introducing `inf2` runtime blog post

### PR DESCRIPTION
**Pull request by spillai**
## Summary

- Added blog post for inf2 runtime
- Updated README with cleaner API / usage and value-props

## Checks

- [ ] `make lint`: I've run `make lint` to lint the changes in this PR.
- [ ] `make test`: I've made sure the tests (`make test-cpu` or `make test`) are passing.
- Additional tests:
   - [ ] Benchmark tests (when contributing new models)
   - [ ] GPU/HW tests

### CODE REVIEW

File: docs/blog/posts/04-introducing-inf2-runtime.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+## 🚀 What is AWS Inferentia2?
+
+[AWS Inferentia2](https://aws.amazon.com/en/ec2/instance-types/inf2/) (Inf2 for short) is the second-generation inference accelerator from AWS. Inf2 instances raise the performance of Inf1 (originally launched in 2019) by delivering 3x higher compute performance, 4x larger total accelerator memory, up to 4x higher throughput, and up to 10x lower latency. Inf2 instances are the first inference-optimized instances in Amazon EC2 to support scale-out distributed inference with ultra-high-speed connectivity between accelerators. 
+
+Relative to the [AWS G5 instances](https://aws.amazon.com/ec2/instance-types/g5/) ([NVIDIA A10G](https://www.nvidia.com/en-us/data-center/products/a10-gpu/)), Inf2 instances promise up to 50% better performance-per-watt. Inf2 instances are ideal for applications such as natural language processing, recommender systems, image classification and recognition, speech recognition, and language translation that can take advantage of scale-out distributed inference.
```
Comment (outtanames) [inline docs/blog/posts/04-introducing-inf2-runtime.md:27]: should we quote the inf2 numbers here against say an A100 for reference?

File: docs/blog/posts/04-introducing-inf2-runtime.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+| inf2.24xlarge | 6                 | 192                     | 96   | 384          | $6.49           | $2.45       |
+| inf2.48xlarge | 12                | 384                     | 192  | 768          | $12.98          | $5.13       |
+
+
+## 📦 Deploying a model on Inferentia2 with NOS
+
+Deploying models on AWS Inferentia2 chips presents a unique set of challenges, distinctly different from the experience with NVIDIA GPUs. This is primarily due to the lack of a mature toolchain for compiling, profiling, and deploying models onto these specialized ASICs. To effectively utilize the AWS Inferentia2 chips, custom model tracing and compilation are essential steps. This process demands a deep understanding of the deployment toolchain, including PyTorch IR op-support and the [AWS Neuron SDK](https://github.com/aws-neuron/aws-neuron-sdk), to optimize model performance fully. NOS aims to bridge this gap and streamline the deployment process, making it more accessible for developers to leverage the powerful inference capabilities of AWS Inferentia2 for their inference workloads and expose easy-to-use gRPC/RESTful services in a straightforward manner.
```
Comment (outtanames) [inline docs/blog/posts/04-introducing-inf2-runtime.md:39]: [nit] distinctly different -> distinct. Also wouldn't say 'lack of mature toolchain', maybe just point out that the Neuron SDK is very differnt from the torch cuda ecosystem.

File: docs/blog/posts/04-introducing-inf2-runtime.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+
+## 🤑 What's it going to cost me?
+
+The table below shows the costs of deploying one of these embedding services on a single Inf2 instance on AWS. While the costs are only one part of the equation, it is important to note that the AWS Inf2 instances are ~25% cheaper than the NVIDIA A10G instances, and offer a more cost-effective solution for inference workloads. In the coming weeks, we'll be digging into evaluating the performance of the Inf2 instances with respect to their NVIDIA GPU counterparts on inference metrics such as latency/throughput etc.
+
+| Model | Cloud Instance | Spot | Cost / hr | Cost / month | # of Req. / $ | 
+| ----- | -------------- | ---- | --------- | ------------ | ---------- |
+| [BAAI/bge-small-en-v1.5](https://huggingface.co/BAAI/bge-small-en-v1.5) | `inf2.xlarge` | - | $0.75 | ~$540 | ~685K / $1 | 
+| **[BAAI/bge-small-en-v1.5](https://huggingface.co/BAAI/bge-small-en-v1.5)** | `inf2.xlarge` | ✅ | **$0.32** | **~$230** | ~1.6M / $1 |
```
Comment (outtanames) [inline docs/blog/posts/04-introducing-inf2-runtime.md:254]: 💯

**Review by outtanames — APPROVED** [review]
Looks Good.

File: docs/blog/posts/04-introducing-inf2-runtime.md (hunk 1 above)
Comment (spillai) [reply docs/blog/posts/04-introducing-inf2-runtime.md:27]: Will do this once we have the profiling numbers to compare, I could only find this stat relative to G5.

**Pull request merged**